server/src/main/java/io/druid/server/lookup/cache/LookupCoordinatorManager.java

```diff
@@ -640,6 +414,258 @@ public void stop()
+      LookupsStateWithMap lookupsUpdate
+  )
+      throws IOException, InterruptedException, ExecutionException
+  {
+    final AtomicInteger returnCode = new AtomicInteger(0);
+    final AtomicReference<String> reasonString = new AtomicReference<>(null);
+
+    final URL url = getLookupsUpdateURL(node);
+
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("Sending lookups load/drop request to [%s]. Request [%s]", url, lookupsUpdate);
+    }
+
+    try (final InputStream result = httpClient.go(
+        new Request(HttpMethod.POST, url)
+            .addHeader(HttpHeaders.Names.ACCEPT, SmileMediaTypes.APPLICATION_JACKSON_SMILE)
+            .addHeader(HttpHeaders.Names.CONTENT_TYPE, SmileMediaTypes.APPLICATION_JACKSON_SMILE)
+            .setContent(smileMapper.writeValueAsBytes(lookupsUpdate)),
+        makeResponseHandler(returnCode, reasonString),
+        lookupCoordinatorManagerConfig.getHostTimeout()
+    ).get()) {
+      if (!httpStatusIsSuccess(returnCode.get())) {
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        try {
+          StreamUtils.copyAndClose(result, baos);
+        }
+        catch (IOException e2) {
+          LOG.warn(e2, "Error reading response");
+        }
+
+        throw new IOException(
+            String.format(
+                "Bad update request to [%s] : [%d] : [%s]  Response: [%s]",
+                url,
+                returnCode.get(),
+                reasonString.get(),
+                StringUtils.fromUtf8(baos.toByteArray())
+            )
+        );
+      } else {
+        try {
+          final LookupsStateWithMap response = smileMapper.readValue(result, LookupsStateWithMap.class);
+          if (LOG.isDebugEnabled()) {
+            LOG.debug(
+                "Update on [%s], Status: %s reason: [%s], Response [%s].", url, returnCode.get(), reasonString.get(),
+                response
+            );
+          }
+          return response;
+        } catch (IOException ex) {
+          throw new IOException(
+              String.format("Failed to parse update response from [%s]. response [%s]", url, result),
+              ex
+          );
+        }
+      }
+    }
+  }
+
+  @VisibleForTesting
+  LookupsStateWithMap getLookupStateForNode(
+      HostAndPort node
+  ) throws IOException, InterruptedException, ExecutionException
+  {
+    final URL url = getLookupsURL(node);
+    final AtomicInteger returnCode = new AtomicInteger(0);
+    final AtomicReference<String> reasonString = new AtomicReference<>(null);
+
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("Getting lookups from [%s]", url);
+    }
+
+    try (final InputStream result = httpClient.go(
+        new Request(HttpMethod.GET, url)
+            .addHeader(HttpHeaders.Names.ACCEPT, SmileMediaTypes.APPLICATION_JACKSON_SMILE),
+        makeResponseHandler(returnCode, reasonString),
+        lookupCoordinatorManagerConfig.getHostTimeout()
+    ).get()) {
+      if (returnCode.get() == 200) {
```

Comment:
that method returns true for anything 2xx while in this GET case only 200 is acceptable.